src/NPadding.cpp

```diff
@@ -125,33 +140,37 @@ double NPadding::NPaddingProb(
   assert(flexbounds.first <= emission_indices.size() &&
          flexbounds.second <= emission_indices.size());
 
-  int g_l, g_r, pad_start, pad_end, n_exclude_count;
-  g_l = flexbounds.first;
-  g_r = flexbounds.second;
-  double prob = (1 - n_transition_prob_);
+  int flex_pad_start, flex_pad_end, n_exclude_count;
+  int g_l = flexbounds.first;
+  int g_r = flexbounds.second;
+  double prob = 1.0;
 
-  // To understand the different cases of the NPadding calculation below,
-  // see https://github.com/matsengrp/linearham/issues/35#issuecomment-270037356.
+  // To understand the different cases of the NPadding calculation below, see
+  // https://github.com/matsengrp/linearham/issues/35#issuecomment-270037356.
 
   // Find the appropriate amount of padding required.
   if (pad_left) {
-    pad_start = g_l;
-    pad_end = std::max(read_pos, g_l);
-    n_exclude_count = std::min(pad_end - read_pos, n_read_count);
+    flex_pad_start = g_l;
+    flex_pad_end = std::max(read_pos, g_l);
+    n_exclude_count = std::min(flex_pad_end - read_pos, n_read_count);
   } else {
-    pad_start = std::min(read_pos, g_r);
-    pad_end = g_r;
-    n_exclude_count = std::min(read_pos - pad_start, n_read_count);
+    flex_pad_start = std::min(read_pos, g_r);
+    flex_pad_end = g_r;
+    n_exclude_count = std::min(read_pos - flex_pad_start, n_read_count);
   }
```

Comment:
See https://github.com/matsengrp/linearham/issues/35#issuecomment-270037356 for a diagram on NPadding possibilities.